server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -1746,11 +1750,11 @@ public boolean canCache(ShardSearchRequest request, SearchContext context) {
         IndexSettings settings = context.indexShard().indexSettings();
         // if not explicitly set in the request, use the index setting, if not, use the request
         if (request.requestCache() == null) {
-            if (settings.getValue(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING) == false) {
-                return false;
-            } else if (context.size() != 0) {
+            if (settings.getValue(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING) == false
```

Comment:
Considering this setting works in tandem with request level override `?request_cache=true`, we need to use it accordingly.
As request level override takes precedence over index level setting(INDEX_CACHE_REQUEST_ENABLED_SETTING), we should have the same behaviour for this cluster setting as well. I think we should do below?



```
 if (request.requestCache() == null || !canCacheSizeNonzeroRequests) {
            if (settings.getValue(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING) == false) {
                return false;
            } else if (context.size() != 0) {
                return false;
            }
        } else if (request.requestCache() == false) {
            return false;
        }
```

Reply:
Ignore this one. I think lets not override index level setting and keep this logic as it is.